proc_Hahn_echoph.py

```diff
@@ -1,67 +1,45 @@
 from pyspecdata import *
 from scipy.optimize import leastsq,minimize,basinhopping
-fl = figlist_var()
+from proc_scripts import *
+from proc_scripts.load_data import postproc_lookup
+from sympy import symbols
+fl = fl_mod()
 t2 = symbols('t2')
 filter_bandwidth = 5e3
-for date,id_string,label_str in [
-        ('191007','echo_2','n'),
+color_choice = True
+for searchstr,exp_type,nodename,postproc,label_str,color_str in [
+        ('200302_alex_probe_water','test_equip','signal','Hahn_echoph','microwaves off','blue'),
         ]:
-    filename = date+'_'+id_string+'.h5'
-    nodename = 'signal'
-    s = nddata_hdf5(filename+'/'+nodename,
-            directory = getDATADIR(
-                exp_type = 'test_equip'))
-    nPoints = s.get_prop('acq_params')['nPoints']
-    nEchoes = s.get_prop('acq_params')['nEchoes']
-    nPhaseSteps = 8 
-    SW_kHz = s.get_prop('acq_params')['SW_kHz']
-    nScans = s.get_prop('acq_params')['nScans']
-    print(ndshape(s))
-    s.reorder('t',first=True)
-    t2_axis = s.getaxis('t')[0:2048]
-    s.setaxis('t',None)
-    s.chunk('t',['ph2','ph1','t2'],[2,4,-1])
-    s.setaxis('ph2',r_[0.,2.]/4)
-    s.setaxis('ph1',r_[0.,1.,2.,3.]/4)
-    s.set_units('t2','s')
-    s.reorder('t2',first=False)
-    s.ft('t2',shift=True)
-    fl.next('ft')
-    fl.plot(s)
-    s = s['t2':(-2e3,2e3)]
-    s.ift('t2')
-    fl.next('raw data')
-    fl.image(s)
-    fl.next('raw data -- coherence pathways')
-    #s = s['power',0].C
-    s.ft(['ph2','ph1'])
-    #s *= exp(-s.getaxis('t2')/10e-3)
-    fl.image(s)
-    fl.show();quit()
-    fl.next('filtered + rough centered data')
-    s.ft('t2', shift=True)
-    s *= exp(-1j*2*pi*10.3)
-    s = s['t2':(-filter_bandwidth/2,filter_bandwidth/2)]
+    s = find_file(searchstr, exp_type=exp_type, expno=nodename,
+            postproc=postproc, lookup=postproc_lookup)
+    s.mean('nScans')    
+    fl.next('signal')
+    slice_f = (-5e3,5e3)
```

Comment:
this is a hard-coded choice -- I prefer to put things like this up at the top.  If it might possibly vary for different datasets, include it in your list of tuples (when we convert to a function, things like this would become optional arguments).

(Apodization is when you multiply by a function that has different complex amplitude for different times.  These functions have a complex amplitude of 1 for all times)

Reply:
I believe this slice is pretty average for most hahn echo data, this is a good range that almost always includes the peak we want. However, I've only worked on processing a few data sets so I may be wrong. I did move it to the top though